### PR TITLE
1-Init-repository

### LINKED ISSUE #1: Initialise repository
- [x] Create sample console app with unit tests
- [x] Add cmake
- [x] Add GitHub jobs
- [x] Add code formatting

### PR DESCRIPTION
**Pull request by Willu12**
init repo with sample hello world
created default cmake to run project
added sample tests

added github actions to check for formatting and tests

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,15 @@
+project(jbcc)
```
Comment (jkurdek) [inline CMakeLists.txt]: Shouldn't `project(jbcc)` come after [cmake_minimum_required()](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html#command:cmake_minimum_required)?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
 cmake_minimum_required(VERSION 3.21)
 
 set(CMAKE_CXX_STANDARD 23)
```
Comment (jkurdek) [inline CMakeLists.txt:4]:
```suggestion
set(CMAKE_CXX_STANDARD 23)
set(CMAKE_CXX_STANDARD_REQUIRED True)
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
 
+add_compile_options(-Wall -Wextra -Wpedantic)
+
+
 set(USE_CCACHE YES)
```
Comment (jkurdek) [inline CMakeLists.txt:11]: What does it do?

Comment (Willu12) [reply CMakeLists.txt:11]: It gives more warnings on compile time, checks for loops that are ale always false, cheks for memory leaks etc.

-Wall is classic one i always use in c
- Wextra and  Wpedantic are ones I wanted to checkout

File: CMakeLists.txt (hunk 1 above)
Comment (Willu12) [reply CMakeLists.txt]: It turns out it should

**Review by jkurdek — APPROVED** [review]
LGTM!

**Pull request merged**